src/main/java/dev/binclub/javaception/runtime/ExecutionEngine.java

```diff
@@ -22,20 +32,696 @@ public static Object invokeMethodObj(InstanceOop instance, MethodInfo method, Ob
+				break;
+			case FCONST_0:
+			case FCONST_1:
+			case FCONST_2:
+				methodContext.push((float) (opcode - FCONST_0));
+				break;
+			case DCONST_0:
+			case DCONST_1:
+				methodContext.push((double) (opcode - DCONST_0));
+				break;
+			case BIPUSH:
+				methodContext.push((int) (instructions[currentInstruction + 1]));
+				break;
+			case SIPUSH:
+				methodContext.push((instructions[currentInstruction + 1] << 8) | instructions[currentInstruction + 2]);
+			case LDC:
+				//todo
+				break;
+			case LDC_W:
+				//todo
+				break;
+			case LDC2_W:
+				//todo
+				break;
+			case ILOAD:
+			case LLOAD:
+			case FLOAD:
+			case DLOAD:
+			case ALOAD:
+				index = ByteUtils.readUnsignedByte(instructions, currentInstruction + 1);
+				methodContext.push(methodContext.load(index));
+				break;
+			case ILOAD_0:
+			case ILOAD_1:
+			case ILOAD_2:
+			case ILOAD_3:
+			case LLOAD_0:
+			case LLOAD_1:
+			case LLOAD_2:
+			case LLOAD_3:
+			case FLOAD_0:
+			case FLOAD_1:
+			case FLOAD_2:
+			case FLOAD_3:
+			case DLOAD_0:
+			case DLOAD_1:
+			case DLOAD_2:
+			case DLOAD_3:
+			case ALOAD_0:
+			case ALOAD_1:
+			case ALOAD_2:
+			case ALOAD_3:
+				methodContext.push(methodContext.load((opcode + 2) % 4));
+				break;
+			case IALOAD:
+			case LALOAD:
+			case FALOAD:
+			case DALOAD:
+			case AALOAD:
+			case BALOAD:
+			case CALOAD:
+			case SALOAD:
+				index = (int) methodContext.pop();
+				Object[] arr = (Object[]) methodContext.pop();
+				methodContext.push(arr[index]);
+				break;
+			case ISTORE:
+			case LSTORE:
+			case FSTORE:
+			case DSTORE:
+			case ASTORE:
+				index = ByteUtils.readUnsignedByte(instructions, currentInstruction + 1);
+				methodContext.store(index, methodContext.pop());
+				break;
+			case ISTORE_0:
+			case ISTORE_1:
+			case ISTORE_2:
+			case ISTORE_3:
+			case LSTORE_0:
+			case LSTORE_1:
+			case LSTORE_2:
+			case LSTORE_3:
+			case FSTORE_0:
+			case FSTORE_1:
+			case FSTORE_2:
+			case FSTORE_3:
+			case DSTORE_0:
+			case DSTORE_1:
+			case DSTORE_2:
+			case DSTORE_3:
+			case ASTORE_0:
+			case ASTORE_1:
+			case ASTORE_2:
+			case ASTORE_3:
+				methodContext.store((opcode + 1) % 4, methodContext.pop());
+				break;
+			case IASTORE:
+			case LASTORE:
+			case FASTORE:
+			case DASTORE:
+			case AASTORE:
+			case BASTORE:
+			case CASTORE:
+			case SASTORE:
+				Object val = methodContext.pop();
+				index = (int) methodContext.pop();
+				arr = (Object[]) methodContext.pop();
+				arr[index] = val;
+				break;
+			case POP:
+				methodContext.pop();
+				break;
+			case POP2:
+				methodContext.pop();
+				methodContext.pop();
+				break;
+			case DUP:
+				val = methodContext.pop();
+				methodContext.push(val);
+				methodContext.push(val);
+				break;
+			case DUP_X1:
+				val = methodContext.pop();
+				Object val2 = methodContext.pop();
+				methodContext.push(val);
+				methodContext.push(val2);
+				methodContext.push(val);
+				break;
+			case DUP_X2:
```

Comment:
These should respect different behaviours based on the category of the stack values (see https://docs.oracle.com/javase/specs/jvms/se15/html/jvms-6.html#jvms-6.5.dup_x2 and more)

Reply:
typo